Speed up and bug fixes

JHanselman opened the pull request:
- Sped up Riemann surface computations a bit
- Sped up differential form computation
- Several bug fixes
- It is no longer needed to choose an embedding when defining a Riemann surface
- Reverted add_row! to add_row (resp. add_column! to add_column) because it produces the wrong results for some reason.

JHanselman committed:
Speed up of differential computation

JHanselman committed:
Merge branch 'master' of https://github.com/JHanselman/Hecke.jl

JHanselman committed:
Fixes. Now it should be possible to define Riemann surfaces without
choosing a valuation.

JHanselman committed:
Merge branch 'thofma:master' into master

JHanselman committed:
Type stability

JHanselman committed:
Merge branch 'thofma:master' into RS_type_stab

JHanselman committed:
Adapted test

JHanselman committed:
Revert add_row! to add_row

simonbrandhorst commented on src/RieSrf/RiemannSurface.jl:
```diff
@@ -1119,17 +1139,17 @@ function symplectic_reduction(K::ZZMatrix)
-        add_row!(B, v, pivot_plus, j)
+        A = add_row(A, v, pivot_plus, j)
+        A = add_column(A, v, pivot_plus, j)
+        B = add_row(B, v, pivot_plus, j)
```
Could you please investigate? If there is something fishy we should know what is wrong.

simonbrandhorst replied:
I cannot reproduce this behavior. Maybe you had an outdated Nemo version?
In some recent version the indices were inconsistent with the doc and abstract algebra 
We had that fixed.

JHanselman committed:
Corrected comment

JHanselman committed:
Added multiplication of function field element with order to produce fractional ideal

JHanselman committed:
Update RiemannSurface.jl

JHanselman committed:
Change test in Ideal.jl 

Changed the test because a*O will now automatically be a fractional ideal if a is an element of a function field.

JHanselman committed:
Correct integration matrix of reverse paths

JHanselman committed:
Merge remote-tracking branch 'refs/remotes/origin/RS_type_stab' into RS_type_stab

JHanselman committed:
Added heuristic ellipse bound for speed. Changed add_row/add_column back as the new Julia version gave problems and seemed to work with the in-place versions again

JHanselman committed:
Added and corrected test

simonbrandhorst reviewed (approved):
Looks good to me.
